lib/thrift/exceptions.ex

```diff
@@ -47,6 +47,15 @@ defmodule Thrift.TApplicationException do
   defp normalize_type(type) when is_integer(type), do: :unknown
 end
 
+defmodule Thrift.ConnectionError do
+  @enforce_keys [:reason]
+  defexception [:reason]
+
+  def message(%{reason: reason}) do
+    "Connection error: #{:inet.format_error(reason)} (#{reason})"
```

Comment:
Might want to special case `:closed` and `:timeout`. They come out of `Thrift.Binary.Framed.Client` but are not POSIX errors.